src/wallet/wallet.cpp

```diff
@@ -2466,12 +2461,8 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             if (coin.m_input_bytes <= 0) {
                 return false; // Not solvable, can't estimate size for fee
             }
-            coin.effective_value = coin.txout.nValue - coin_selection_params.m_effective_feerate.GetFee(coin.m_input_bytes);
-            if (coin_selection_params.use_bnb) {
-                value_to_select -= coin.effective_value;
-            } else {
-                value_to_select -= coin.txout.nValue;
-            }
+            coin.effective_value = coin.txout.nValue - (coin_selection_params.m_subtract_fee_outputs ? 0 : coin_selection_params.m_effective_feerate.GetFee(coin.m_input_bytes));
```

Comment:
I think it may be good if `CInputCoin` kept track of both `nValue` and `n_effective_value` separately. In the case of recipient paying for the tx, we could use effective value for filtering, but `nValue` to determine whether sufficient funds have been selected.

Reply:
Then also, `effective_value` wouldn't need to be instantiated with `nValue` which I thought was pretty surprising.

Reply:
This particular section is for pre-selected inputs, so we don't filter here. However I do agree that changing the effective_feerate depending on whether we are subtracting the fee from the outputs doesn't quite make sense. So I've changed this back to subtract the correct value from `value_to_select` based on `coin_selection_params.m_subtract_fee_outputs`.

> Then also, `effective_value` wouldn't need to be instantiated with `nValue` which I thought was pretty surprising.

I don't think that is necessarily true.